### PR TITLE
Vanilla Docs update

### LINKED ISSUE #1205: Documentation should link to design spec

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Standardised heading sizes for docs pages to h2 (with an off-screen h1 at the top)
- Updated the header nav and side nav to Vanilla styling (1.6.4)
- Added GitHub and Report a bug links to component headers
- Added links to design spec for each component
- Added the missing Aside pattern to `metadata.yaml` under Navigation
- Removed TOC on small screens - will be addressed via #1531 

## QA

- Pull code
- Run `cd docs && ./run`
- Open http://0.0.0.0:8104/
- Check that the Header and Side navs use Vanilla styling (reference [here](https://vanilla-framework.github.io/vanilla-framework/examples/patterns/navigation/default/) and [here](https://vanilla-framework.github.io/vanilla-framework/examples/patterns/navigation/sidebar/))
- Check that there are links for GitHub and Report a bug on the component headers (reference [here](https://github.com/ubuntudesign/vanilla-design/issues/119))
- Check that each pattern has a h2 heading
- Check that each pattern has a link to the relevant design spec (if it exists) at the bottom
- Check that the Aside pattern link exists under the Navigation subheading

## Details

Fixes ubuntudesign/docs.vanillaframework.io#115
Fixes ubuntudesign/docs.vanillaframework.io#116
Fixes ubuntudesign/docs.vanillaframework.io#118
Fixes #1205 

## Screenshots
screenshot sdfsdf

### CODE REVIEW

File: docs/en/base/code.md [hunk 1]
```diff
@@ -3,15 +3,15 @@ title: Code
 table_of_contents: true
 ---
 
-# Code
+## Code
```
Comment (anthonydillon) [inline docs/en/base/code.md]: Semantically the first title of the pages should be h1's

**Comment by Caleb-Ellis** [comment]
@kwm14 Updated with your suggestions, see screenshot of how it looks now

@anthonydillon I'm not sure what the best way to get h2 styling on h1s would be. I tried appending the `.p-heading--two` etc classes to the heading tags using javascript, but it causes a moment of flashing from h1 -> h2 styling. For now I've got a super naive solution of changing the heading styles in the css which solves the flashing but.... it feels wrong. Any suggestions?

screenshot sdfsdf

**Comment by kwm14** [comment]
@Caleb-Ellis thanks for the screenshots. Design updates look great 👍

**Comment by anthonydillon** [comment]
@Caleb-Ellis we should have a h1 on the page so let's add a title of Vanilla documentation as a h1 and off screen it. Then the visible page titles can be h2s.

File: docs/template.html [hunk 2]
```diff
@@ -238,287 +204,163 @@
...
-                  <li class="p-versions__item">
-                    <a href="{{ version.path }}" class="p-versions__link">{{ version.name }}</a>
-                  </li>
-                  {% endif %}
-                {% endfor %}
+      </div>
+      {% endif %}
+      <div class="col-8 main-col u-no-margin--top">
+        <main id="main-content">
+          <h1 class="u-off-screen">Vanilla Framework Documentation</h1>
```
Comment (anthonydillon) [inline docs/template.html]: Please update to `Vanilla framework documentation`

**Comment by anthonydillon** [comment]
## QA issues
- The new header links are not responsive:
screenshot-2018-1-23 breadcrumbs documentation

**Comment by kwm14** [comment]
Looknig good @Caleb-Ellis

**Pull request merged**